dd-java-agent/instrumentation/trace-annotation/src/main/java/datadog/trace/instrumentation/trace_annotation/TraceAdvice.java

```diff
@@ -18,24 +18,22 @@
 
   @Advice.OnMethodEnter(suppress = Throwable.class)
   public static AgentScope onEnter(@Advice.Origin final Method method) {
-    Annotation trace = method.getAnnotation(Trace.class);
-    if (trace == null) {
-      trace = method.getAnnotation(com.signalfx.tracing.api.Trace.class);
-    }
-
-    boolean useResourceAsOperation = false;
     String operationName = null;
     String resourceName = null;
+    boolean useResourceAsOperation = true;
```

Comment:
The logic of when the resource is used as an operation name is not entirely clear. Is there a way to refactor this to make more understandable? I have a hard time following all possible code paths where useResourceAsOperation, operationName and resourceName are modified.

Reply:
@tigrannajaryan is this better?  I removed the flag and opted for saner defaults.

Reply:
Much better, thank you.